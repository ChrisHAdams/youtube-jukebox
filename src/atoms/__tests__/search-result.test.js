import React from "react"
import renderer from "react-test-renderer"
import SearchResult from "../search-result";

import {render, fireEvent, cleanup} from '@testing-library/react'
import {toHaveStyle} from '@testing-library/jest-dom';

expect.extend({toHaveStyle})

const videoMock = {
  id: "someId",
  title: "Some Title",
  thumbnails:{
    default:{
      url:"www.something"
    }
  }
}


describe("SearchResult", () => {

  afterEach(cleanup);

  it("renders correctly", () => {

    const onClick = (video)=>{
      console.log("New search value " + video.id);
    }

    const tree = renderer
      .create(<SearchResult
                video={videoMock}
                lightTheme={false}
                onClick={onClick}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })


  it('calls "onClick" prop on button click', () => {

    // Render new instance in every test to prevent leaking state
    //const onClick = jest.fn();
    const lightThemeValue = false;
    let returnValue = '';

    const onClickFunc = (video)=>{
      returnValue = video.id;
    }

    const { getByTestId } = render(<SearchResult
                                      video={videoMock}
                                      lightTheme={lightThemeValue}
                                      onClick={onClickFunc}/>);

    fireEvent.click(getByTestId('searchResultImage'));
    expect(videoMock.id).toEqual(returnValue);

    expect(getByTestId('searchResult')).toHaveStyle(`background-color: '#293447'`);

  });

  it('calls "onClick" prop on button click 2', () => {

    // Render new instance in every test to prevent leaking state
    //const onClick = jest.fn();
    const lightThemeValue = true;


    const onClickFunc = (video)=>{
      let returnValue = video.id;
    }

    const { getByTestId } = render(<SearchResult
                                      video={videoMock}
                                      lightTheme={lightThemeValue}
                                      onClick={onClickFunc}/>);


    expect(getByTestId('searchResult')).toHaveStyle(`background-color: '#a7aebc'`);  //Light Value


  });
})


