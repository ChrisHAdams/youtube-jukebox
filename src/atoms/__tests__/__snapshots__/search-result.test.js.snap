// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders correctly 1`] = `
<div>
  <div
    className="sc-htpNat dAUblb"
    data-testid="searchResult"
  >
    <img
      alt="Some Title"
      className="sc-ifAKCX eBOhBv"
      data-testid="searchResultImage"
      onClick={[Function]}
      src="www.something"
      title="Some Title"
    />
    <p
      className="sc-bxivhb jhgCzO"
    >
      Some Title
      ...
    </p>
  </div>
</div>
`;
